lib/elixir/lib/enum.ex

```diff
@@ -2390,6 +2396,12 @@ defmodule Enum do
     reduce_range(first, last, step, acc, fun)
   end
 
+  # TODO: Remove me on v2.0
+  def reduce(%{__struct__: Range, first: first, last: last} = range, acc, fun) do
```

Comment:
I believe we don't need this clause because it will fallback to the protocol.

Reply:
You're right
```
iex(6)> Enum.reduce(old_range, 0, fn a, b -> a + b end)
55
```